lib/spack/spack/test/cmd/env.py

```diff
@@ -2403,6 +2403,28 @@ def test_env_activate_default_view_root_unconditional(mutable_mock_env_path):
     )
 
 
+def test_env_activate_custom_view(tmpdir, install_mockery, mock_fetch):
+    """Check that an environment can be activated with a non-default view."""
+    filename = str(tmpdir.join("spack.yaml"))
```

Comment:
Then you can use the Path-like object:
```suggestion
    filename = tmp_path / "spack.yaml"
```